### PR TITLE
sql/parse: fix order by with group by

### LINKED ISSUE #228: Order by does not work with group by
More info: https://github.com/src-d/gitbase/issues/303#issuecomment-398110315

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #228

### CODE REVIEW

**Comment by ajnavarro** [comment]
The order appear to change: https://travis-ci.org/src-d/go-mysql-server/jobs/394033752#L731

**Comment by erizocosmico** [comment]
Um, yes, because the count is the same in both and the sort we're doing must not be stable, damn

**Comment by erizocosmico** [comment]
Fixed @ajnavarro

**Comment by ajnavarro** [comment]
Same error: https://travis-ci.org/src-d/go-mysql-server/jobs/394045184#L729

**Comment by erizocosmico** [comment]
Should be fixed now

File: engine_test.go [hunk 1]
```diff
@@ -689,32 +726,7 @@ func TestTracing(t *testing.T) {
...
-		if strings.HasPrefix(s, "plan.") ||
-			strings.HasPrefix(s, "expression.") ||
-			strings.HasPrefix(s, "function.") ||
-			strings.HasPrefix(s, "aggregation.") {
-			spanOperations = append(spanOperations, s)
-		}
-	}
-
-	require.Equal(expectedSpans, spanOperations)
+	require.True(len(tracer.spans) >= 0)
```
Comment (ajnavarro) [inline engine_test.go]: I would prefer to test the expected spans rather than there are some of them.

**Comment by erizocosmico** [comment]
Done @ajnavarro

**Comment by erizocosmico** [comment]
Rebased. PTAL @kuba--

**Comment by mcarmonaa** [comment]
CI is failing:
```
# gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa
sql/index/pilosa/driver_test.go:431:33: cannot use context.Background() (type context.Context) as type *sql.Context in argument to d.Save: need type assertion
FAIL	gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa [build failed]
```

Is it still pending of rebase?

**Comment by erizocosmico** [comment]
@mcarmonaa yes, something was merged after that rebase

**Comment by erizocosmico** [comment]
Fixed

**Pull request merged**